crates/chain/src/indexed_tx_graph.rs

```diff
@@ -252,61 +252,62 @@ where
     }
 }
 
-/// Methods are available if the anchor (`A`) implements [`AnchorFromBlockPosition`].
-impl<A: Anchor, I: Indexer> IndexedTxGraph<A, I>
+/// Methods are available if the anchor (`A`) can be created from [`BlockPosition`].
+impl<'b, A, I> IndexedTxGraph<A, I>
 where
     I::ChangeSet: Default + Merge,
-    A: AnchorFromBlockPosition,
+    A: Anchor + From<BlockPosition<'b>>,
```

Comment:
Maybe instead of adding a new type this can be `A: Anchor + From<ConfirmationBlockTime>` and below we just construct a `ConfirmationBlockTime` from a block id and header time.

My other thought is that `BlockPosition` might not be the right name if originally it referred to the position of the txid in the `txdata` vector, but that could be up to interpretation.

Reply:
@ValuedMammal The point of the type constraint is that you can get the anchor type from a block and height. We don't want to constrain it so that it must come from a `ConfirmationBlockTime` since not all anchors can do that -- they can all come from a full block + height though.

I agree about the name.

I think the constraint should be `A: for<'b> From<BlockPosition<'b>>` since it should be from any lifetime.

Reply:
> I think the constraint should be A: for<'b> From<BlockPosition<'b>> since it should be from any lifetime.

Good idea!